File: src/Models/ApiKey.cs
namespace mithrandir.Models;

public class ApiKey
{
  public int Id { get; set; }
  public required string KeyHash { get; set; }
  public required string Name { get; set; }
  public Tier Tier { get; set; }
  public Status Status { get; set; }
  public string[]? IpWhitelist { get; set; }
  public DateTimeOffset CreatedAt { get; set; }
  public DateTimeOffset? ExpiresAt { get; set; }
  public DateTimeOffset? LastUsedAt { get; set; }
}

File: dashboard/types/api-key.ts
import { Status, Tier } from "./enums";

export interface ApiKey {
  id: number;
  keyHash?: string;
  name: string;
  tier: Tier;
  status: Status;
  ipWhitelist: string[] | null;
  createdAt: string;
  expiresAt: string | null;
  lastUsedAt: string | null;
}
